templates/CRM/common/paymentBlock.tpl

```diff
@@ -108,15 +111,15 @@
           }
         }
       }
-      dataUrl =  dataUrl + "&payment_instrument_id=" + payment_instrument_id;
+      dataUrl =  dataUrl + "&payment_instrument_id=" + payment_instrument_id + "&currency=" + currency;
 
       // Processors like pp-express will hide the form submit buttons, so re-show them when switching
       $('.crm-submit-buttons', $form).show().find('input').prop('disabled', true);
       CRM.loadPage(dataUrl, {target: '#billing-payment-block'});
     }
 
-    $('[name=payment_processor_id]').on('change.paymentBlock', function() {
-        buildPaymentBlock($(this).val());
+    $('[name=payment_processor_id], #currency').on('change.paymentBlock', function() {
+      buildPaymentBlock($('[name=payment_processor_id]').val());
```

Comment:
We need to add `:checked` or it will always take the first option.

Reply:
Tested on a site with a payment processor and pay later and the pay later option was broken.

Reply:
Thanks @samuelsov for raising this issue and suggesting the fix. I have updated the PR with this commit https://github.com/civicrm/civicrm-core/pull/18517/commits/b86dc53eb2c51246435fedcc96b2ba947deea6c7 as per which we need to consider both the w/o pay-later cases.